lib/mumukit/nuntius/connection.rb

```diff
@@ -1,17 +1,24 @@
 class Mumukit::Nuntius::Connection
 
   class << self
-
     def config
       @config ||= YAML.load_interpolated(File.expand_path '../../../../config/rabbit.yml', __FILE__).
           with_indifferent_access[ENV['RAILS_ENV'] || ENV['RACK_ENV'] || 'development']
     end
 
     def start(exchange_name)
-      connection = Bunny.new(host: config[:host], user: config[:user], password: config[:password])
+      raise 'Nuntius connection isn\'t established' if connection.blank?
       channel = connection.start.create_channel
       exchange = channel.fanout(exchange_name)
-      [connection, channel, exchange]
+      [channel, exchange]
+    end
+
+    def establish_connection
+      @@connection = Bunny.new(host: config[:host], user: config[:user], password: config[:password])
+    end
+
+    def connection
+      @@connection ||= nil
```

Comment:
The `||= nil` is not necessary. `nil` is the default value.

Reply:
This avoids checking if the class variable is defined.
I could just use the class variable and ask for definition if you prefer

Reply:
OK, but why aren't we using a plain class-instance variable? `@connection` would just work, right?